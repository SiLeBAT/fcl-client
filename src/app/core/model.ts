export interface ValidationError {
    code: string;
    message: string;
}
