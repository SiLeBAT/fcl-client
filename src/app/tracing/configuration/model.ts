import {
    Color,
    DataServiceInputState,
    LabelPart,
    NodeShapeType,
    TableColumn,
    TableRow,
} from "../data.model";
import { ComplexFilterCondition, PropToValuesMap } from "./configuration.model";

export interface RowFilter<T> {
    filter(arr: TableRow[]): TableRow[];
    getSettings(): T;
}

export interface TextFilterSettings {
    filterTerm: string;
    filterProps: string[];
}

export enum TableType {
    STATIONS,
    DELIVERIES,
}

export type TreeStatus = "collapsed" | "expanded";

export interface EditHighlightingServiceData {
    favouriteProperties: TableColumn[];
    otherProperties: TableColumn[];
    propToValuesMap: PropToValuesMap;
    ruleListItems: RuleListItem[];
}

export interface RuleListItem {
    id: RuleId;
    name: string;
    color: Color | null;
    shape: NodeShapeType | null;
    showInLegend: boolean;
    disabled: boolean;
    autoDisabled: boolean;
    isAnonymizationRule: boolean;
    effElementsCountTooltip: string;
    effElementsCount: number;
    conflictCount: number;
    ruleType: RuleType | null;
}

export interface EditRuleCore {
    id: string;
    name: string;
    userDisabled: boolean;
    complexFilterConditions: ComplexFilterCondition[];
    isValid: boolean;
}

export type EditRule =
    | ColorAndShapeEditRule
    | ColorEditRule
    | LabelEditRule
    | InvEditRule;

export enum RuleType {
    INVISIBILITY,
    LABEL,
    COLOR_AND_SHAPE,
    COLOR,
}

export interface ColorAndShapeEditRule extends EditRuleCore {
    type: RuleType.COLOR_AND_SHAPE;
    showInLegend: boolean;
    color: Color | null;
    shape: NodeShapeType | null;
}

export interface ColorEditRule extends EditRuleCore {
    type: RuleType.COLOR;
    showInLegend: boolean;
    color: Color;
}

export interface SimpleLabelEditRule extends EditRuleCore {
    type: RuleType.LABEL;
    labelProperty: string | null;
}

export interface ComposedLabelEditRule extends EditRuleCore {
    type: RuleType.LABEL;
    labelParts: LabelPart[];
    labelPrefix: string;
}

export type LabelEditRule = SimpleLabelEditRule | ComposedLabelEditRule;

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface InvEditRule extends EditRuleCore {
    type: RuleType.INVISIBILITY;
}

export type StationEditRule =
    | ColorAndShapeEditRule
    | LabelEditRule
    | InvEditRule;
export type DeliveryEditRule = LabelEditRule | InvEditRule | ColorEditRule;

export type DeliveryRuleType = RuleType.LABEL | RuleType.COLOR;
export type StationRuleType = RuleType.LABEL | RuleType.COLOR_AND_SHAPE;

export type RuleId = string;

export interface EditHighlightingState<T extends EditRule> {
    dataServiceInputState: DataServiceInputState;
    editRules: T[];
}
