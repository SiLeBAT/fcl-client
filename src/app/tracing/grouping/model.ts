import {
    StationData, GroupType, GroupData,
    StationTracingSettings, Position, DataServiceInputState
} from '../data.model';

export interface SetStationGroupsPayload {
    groupSettings: GroupData[];
    stationTracingSettings: StationTracingSettings[];
    stationPositions: { [key: string]: Position };
    selectedStations: string[];
    invisibleStations: string[];
}

export interface GroupingChange {
    addGroups: GroupData[];
    removeGroups: GroupData[];
}

export interface GroupingState extends DataServiceInputState {
    stationPositions: { [key: string]: Position };
}

export interface LinkGroup {
    linkStation: StationData;
    linkedStations: {
        linkedStation: StationData;
        linkKeys: string[];
    }[];
}

export type GroupInfoFun = (linkGroup: LinkGroup, groupNumber: number) => { id: string; name: string; groupType: GroupType };

export interface IsolatedComponent {
    ids: string[];
    support: string[];
}
