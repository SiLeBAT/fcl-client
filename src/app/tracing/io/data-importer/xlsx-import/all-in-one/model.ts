import { DeliveryRow, ImportTable, StationRow } from "../model";
import { CellValue } from "../xlsx-reader";

export enum StationColumn {
    EXT_ID,
    NAME,
    STREET,
    STREET_NUMBER,
    ZIP,
    CITY,
    DISTRICT,
    STATE,
    COUNTRY,
    TYPE_OF_BUSINESS,
    ADDCOLS,
}

export enum DeliveryColumn {
    EXT_ID,
    SOURCE,
    PRODUCT_NAME,
    LOT_NUMBER,
    LOT_AMOUNT_NUMBER,
    LOT_AMOUNT_UNIT,
    DATE_OUT_DAY,
    DATE_OUT_MONTH,
    DATE_OUT_YEAR,
    DATE_IN_DAY,
    DATE_IN_MONTH,
    DATE_IN_YEAR,
    UNIT_AMOUNT_NUMBER,
    UNIT_AMOUNT_UNIT,
    TARGET,
    ADDCOLS,
}

export enum Del2DelColumn {
    FROM,
    TO,
}

export interface AllInOneDeliveryRow extends DeliveryRow {
    extId?: string;
    otherProps: Record<string, CellValue>;
}

export interface AllInOneStationRow extends StationRow {
    extId: string;
    typeOfBusiness?: string;
    otherProps: Record<string, CellValue>;
}

export type AllInOneDeliveryTable = ImportTable<AllInOneDeliveryRow>;
