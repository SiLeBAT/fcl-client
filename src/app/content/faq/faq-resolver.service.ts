import { Injectable } from '@angular/core';
import { ActivatedRouteSnapshot, RouterStateSnapshot, Resolve } from '@angular/router';
import { catchError } from 'rxjs/operators';
import { EMPTY, Observable } from 'rxjs';
import { Faq } from './faq.model';
import { FaqService } from './faq.service';
import { AlertService } from '@app/shared/services/alert.service';

@Injectable({
    providedIn: 'root'
})
export class FaqResolverService implements Resolve<Faq> {

    constructor(
        private faqService: FaqService,
        private alertService: AlertService
    ) { }

    resolve(_activatedRoute: ActivatedRouteSnapshot, _snap: RouterStateSnapshot): Observable<Faq> {
        return this.faqService.getFaq().pipe(
            catchError((error) => {
                this.alertService.error(error);
                return EMPTY;
            })
        );
    }
}
