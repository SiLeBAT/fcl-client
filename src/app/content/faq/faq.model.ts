export interface FaqEntry {
  question: string;
  answer: string;
}

export interface FaqSection {
  title: string;
  urlFragment: string;
  entries: FaqEntry[];
}

export interface Faq {
  topEntries: FaqEntry[];
  sections: FaqSection[];
}
