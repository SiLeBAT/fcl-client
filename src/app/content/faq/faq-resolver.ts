import { inject } from "@angular/core";
import {
    ActivatedRouteSnapshot,
    ResolveFn,
    RouterStateSnapshot,
} from "@angular/router";
import { catchError } from "rxjs/operators";
import { EMPTY } from "rxjs";
import { Faq } from "./faq.model";
import { FaqService } from "./faq.service";
import { AlertService } from "@app/shared/services/alert.service";

export const FaqResolverFn: ResolveFn<Faq> = (
    route: ActivatedRouteSnapshot,
    state: RouterStateSnapshot,
) => {
    const faqService = inject(FaqService);
    const alertService = inject(AlertService);

    return faqService.getFaq().pipe(
        catchError((error) => {
            alertService.error(error);
            return EMPTY;
        }),
    );
};
