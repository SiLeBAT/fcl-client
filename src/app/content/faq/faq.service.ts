import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import { Observable } from 'rxjs';
import { Faq, FaqEntry, FaqSection } from './faq.model';
import { ContentService } from '../services/content.service';
import { FaqEntryDTO, FaqResponseDTO, FaqSectionDTO } from '../models/content.model';

@Injectable({
    providedIn: 'root'
})
export class FaqService {

    constructor(private contentService: ContentService) { }

    getFaq(): Observable<Faq> {
        return this.contentService.getFaq().pipe(
            map(dto => this.mapFaq(dto))
        );
    }

    private mapFaq(dto: FaqResponseDTO): Faq {
        return {
            topEntries: (dto.topFaq || []).map(entry => this.mapEntry(entry)),
            sections: dto.sections.map(section => this.mapSection(section))
        };
    }

    private mapSection(dto: FaqSectionDTO): FaqSection {
        return {
            title: dto.title,
            urlFragment: dto.url,
            entries: dto.faq.map(entry => this.mapEntry(entry))
        };
    }

    private mapEntry(dto: FaqEntryDTO): FaqEntry {
        return {
            question: dto.q,
            answer: dto.a
        };
    }
}
