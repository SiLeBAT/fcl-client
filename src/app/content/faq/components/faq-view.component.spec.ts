import { ComponentFixture, TestBed, waitForAsync } from "@angular/core/testing";
import { CUSTOM_ELEMENTS_SCHEMA, SecurityContext } from "@angular/core";

import { FaqViewComponent } from "./faq-view.component";
import { FaqSectionViewComponent } from "./faq-section-view.component";
import { MaterialModule } from "../../../shared/material.module";
import { Faq } from "../faq.model";
import { NoopAnimationsModule } from "@angular/platform-browser/animations";
import { DomSanitizer, SafeValue } from "@angular/platform-browser";
import { RouterTestingModule } from "@angular/router/testing";

describe("FaqViewComponent", () => {
    const mockDomSanitzer: Partial<DomSanitizer> = {
        sanitize: (
            context: SecurityContext,
            value: string | SafeValue | null,
        ) => (value || "") as string,
    };

    const faq: Faq = {
        topEntries: [],
        sections: [
            {
                title: "Section A",
                urlFragment: "sectiona",
                entries: [
                    { question: "Q1", answer: "A1" },
                    { question: "Q2", answer: "A2" },
                ],
            },
            {
                title: "Section B",
                urlFragment: "sectionb",
                entries: [{ question: "Q3", answer: "A1" }],
            },
        ],
    };

    let component: FaqViewComponent;
    let fixture: ComponentFixture<FaqViewComponent>;

    beforeEach(waitForAsync(() => {
        // eslint-disable-next-line @typescript-eslint/no-floating-promises
        TestBed.configureTestingModule({
            declarations: [FaqViewComponent, FaqSectionViewComponent],
            imports: [
                RouterTestingModule,
                MaterialModule,
                NoopAnimationsModule,
            ],
            providers: [
                {
                    provide: DomSanitizer,
                    useValue: mockDomSanitzer as DomSanitizer,
                },
            ],
            schemas: [CUSTOM_ELEMENTS_SCHEMA],
        })
            .compileComponents()
            .then(() => {
                fixture = TestBed.createComponent(FaqViewComponent);
                component = fixture.componentInstance;
                component.faq = faq;
                component.activeFragment = null;
                fixture.detectChanges();
            });
    }));

    it("should create the component", () => {
        expect(component).toBeTruthy();
    });

    it("should render the component correctly", () => {
        expect(fixture).toMatchSnapshot();
    });
});
