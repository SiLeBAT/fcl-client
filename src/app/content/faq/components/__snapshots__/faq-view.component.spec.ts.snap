// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqViewComponent should render the component correctly 1`] = `
<fcl-faq-view
  activeFragment="null"
  faq={[Function Object]}
  faqSectionRefs={[Function QueryList]}
  scrollPanel={[Function ElementRef]}
>
  <div
    class="fcl-scroll-panel"
  >
    <div
      class="fcl-title mat-headline"
    >
      Frequently Asked Questions
    </div>
    <div>
      <hr />
    </div>
    <nav
      class="fcl-toc"
    >
      <a
        class="fcl-toc-link mat-h4"
        href="/#sectiona"
      >
        Section A
      </a>
      <a
        class="fcl-toc-link mat-h4"
        href="/#sectionb"
      >
        Section B
      </a>
    </nav>
    <div
      class="fcl-sections"
    >
      <section>
        <fcl-faq-section-view>
          <mat-accordion
            class="mat-accordion mat-accordion-multi"
            multi="true"
          >
          </mat-accordion>
        </fcl-faq-section-view>
      </section>
      <section
        id="sectiona"
      >
        <fcl-faq-section-view>
          <mat-accordion
            class="mat-accordion mat-accordion-multi"
            multi="true"
          >
            <h3
              class="fcl-title mat-h3"
            >
              Section A
            </h3>
            <mat-expansion-panel
              class="mat-expansion-panel _mat-animation-noopable"
            >
              <mat-expansion-panel-header
                aria-disabled="false"
                aria-expanded="false"
                class="mat-expansion-panel-header mat-focus-indicator mat-expansion-toggle-indicator-after _mat-animation-noopable"
                collapsedheight="auto"
                expandedheight="auto"
                role="button"
                style="height: auto;"
                tabindex="0"
              >
                <span
                  class="mat-content"
                >
                  <mat-panel-title
                    class="mat-expansion-panel-header-title"
                  >
                    <div
                      class="fcl-panel-title"
                    >
                      Q1
                    </div>
                  </mat-panel-title>
                </span>
                <span
                  class="mat-expansion-indicator ng-trigger ng-trigger-indicatorRotate"
                  style="transform:rotate(0deg);0:transform;"
                />
              </mat-expansion-panel-header>
              <div
                class="mat-expansion-panel-content ng-trigger ng-trigger-bodyExpansion"
                role="region"
                style="height:0px;visibility:hidden;0:height;1:visibility;"
              >
                <div
                  class="mat-expansion-panel-body"
                >
                </div>
              </div>
            </mat-expansion-panel>
            <mat-expansion-panel
              class="mat-expansion-panel _mat-animation-noopable"
            >
              <mat-expansion-panel-header
                aria-disabled="false"
                aria-expanded="false"
                class="mat-expansion-panel-header mat-focus-indicator mat-expansion-toggle-indicator-after _mat-animation-noopable"
                collapsedheight="auto"
                expandedheight="auto"
                role="button"
                style="height: auto;"
                tabindex="0"
              >
                <span
                  class="mat-content"
                >
                  <mat-panel-title
                    class="mat-expansion-panel-header-title"
                  >
                    <div
                      class="fcl-panel-title"
                    >
                      Q2
                    </div>
                  </mat-panel-title>
                </span>
                <span
                  class="mat-expansion-indicator ng-trigger ng-trigger-indicatorRotate"
                  style="transform:rotate(0deg);0:transform;"
                />
              </mat-expansion-panel-header>
              <div
                class="mat-expansion-panel-content ng-trigger ng-trigger-bodyExpansion"
                role="region"
                style="height:0px;visibility:hidden;0:height;1:visibility;"
              >
                <div
                  class="mat-expansion-panel-body"
                >
                </div>
              </div>
            </mat-expansion-panel>
          </mat-accordion>
        </fcl-faq-section-view>
      </section>
      <section
        id="sectionb"
      >
        <fcl-faq-section-view>
          <mat-accordion
            class="mat-accordion mat-accordion-multi"
            multi="true"
          >
            <h3
              class="fcl-title mat-h3"
            >
              Section B
            </h3>
            <mat-expansion-panel
              class="mat-expansion-panel _mat-animation-noopable"
            >
              <mat-expansion-panel-header
                aria-disabled="false"
                aria-expanded="false"
                class="mat-expansion-panel-header mat-focus-indicator mat-expansion-toggle-indicator-after _mat-animation-noopable"
                collapsedheight="auto"
                expandedheight="auto"
                role="button"
                style="height: auto;"
                tabindex="0"
              >
                <span
                  class="mat-content"
                >
                  <mat-panel-title
                    class="mat-expansion-panel-header-title"
                  >
                    <div
                      class="fcl-panel-title"
                    >
                      Q3
                    </div>
                  </mat-panel-title>
                </span>
                <span
                  class="mat-expansion-indicator ng-trigger ng-trigger-indicatorRotate"
                  style="transform:rotate(0deg);0:transform;"
                />
              </mat-expansion-panel-header>
              <div
                class="mat-expansion-panel-content ng-trigger ng-trigger-bodyExpansion"
                role="region"
                style="height:0px;visibility:hidden;0:height;1:visibility;"
              >
                <div
                  class="mat-expansion-panel-body"
                >
                </div>
              </div>
            </mat-expansion-panel>
          </mat-accordion>
        </fcl-faq-section-view>
      </section>
    </div>
  </div>
</fcl-faq-view>
`;
